react-forms/Forms/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,34 @@
+import { FormEvent } from 'react';
+import './RegistrationFormUncontrolled.css';
+import { useState } from 'react';
+
+export function RegistrationFormControlled() {
+  const [pass, setPass] = useState('');
+  const [username, setUsername] = useState('');
+
+  function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    console.log('password:', pass);
+    console.log('username:', username);
+  }
+
+  return (
+    <>
+      <form onSubmit={handleSubmit}>
+        <label>
+          Username
+          <input onChange={(e) => setUsername(e.target.value)} type="text" />
```

Comment:
You need to also set `value` on controlled inputs. It doesn't make a difference here, because they default to an empty value, but in general you will need to set the value.